tests/tracking/test_tracking.py

```diff
@@ -1209,3 +1209,16 @@ def save(self, path):
     with mlflow.start_run():
         # Log the dictionary as a table
         mlflow.log_table(data=table_dict, artifact_file=artifact_file)
+
+
+def test_set_async_logging_threadpool_size():
+    MLFLOW_ASYNC_LOGGING_THREADPOOL_SIZE.set(6)
+    assert MLFLOW_ASYNC_LOGGING_THREADPOOL_SIZE.get() == 6
+
+    with mlflow.start_run():
+        mlflow.log_param("key", "val", synchronous=False)
```

Comment:
can we ensure this operation has completed?

Reply:
hmm do we need that? The purpose is to test the threadpool size, and the async queue is automatically flushed on program exit.

Reply:
I think we should. If we do something async, we should wait for that to complete.

Reply:
What if the database is cleaned up before logging completes? This can lead to an error or warning.

Reply:
> What if the database is cleaned up before logging completes?
For this unit test, I don't think python garbage collection will destroy the database since it is held by the async logging queue.

It doesn't hurt though, I am adding the `mlflow.flush_async_logging`.